core/unix/os.c

```diff
@@ -7013,17 +7013,21 @@ pre_system_call(dcontext_t *dcontext)
 #ifdef LINUX
     case SYS_ppoll: {
         kernel_sigset_t *sigmask = (kernel_sigset_t *)sys_param(dcontext, 3);
+        dcontext->sys_param3 = (reg_t)sigmask;
+        if (sigmask == NULL)
+            break;
         size_t sizemask = (size_t)sys_param(dcontext, 4);
-        /* The original app's sigmask parameter is now NULL effectively making the syscall
-         * a non p* version, and the mask's semantics are emulated by DR instead.
+        /* The original app's sigmask parameter is now NULL effectively making the
+         * syscall a non p* version, and the mask's semantics are emulated by DR
+         * instead.
```

Comment:
nit: There are formatting-only changes in these comments which for cleaner history would best be undone: here and below

Reply:
Done.